File: kiyoshi.js
"use strict";
const sleep=require("util").promisify(setTimeout);

(async function(){
	const ptn=["ズン","ドコ"];
	const fin="キ・ヨ・シ!";
	const log=["","","","",""];
	const ans="ズンズンズンズンドコ";

	for(;;){
		let last=ptn[0|Math.random()*ptn.length];
		process.stdout.write(last);
		log.shift();
		log.push(last);
		if(ans==log.join("")) break;
		await sleep(200);
	}
	console.log(fin);
})();

File: kiyoshi.py
import time
import random

if __name__=="__main__":
	ptn=["ズン","ドコ"]
	fin="キ・ヨ・シ!"
	log=["","","","",""]
	ans="ズンズンズンズンドコ"

	while True:
		last=random.choice(ptn)
		print(last,end="",flush=True)
		log.pop(0)
		log.append(last)
		if ans=="".join(log): break
		time.sleep(0.2)

	print(fin)
